src/account-utils.ts

```diff
@@ -58,142 +44,61 @@ export function signMultiCall(
- * @param callParameters array witht the call parameters
- * @param nonce current nonce
- * @returns the message hash for the multicall and the arguments to execute it with
- */
-export function handleMultiCall(
-    accountAddress: string,
-    callParameters: CallParameters[],
-    nonce: Numeric,
-    maxFee: Numeric,
-    version: Numeric
-) {
-    // Transform a CallParameters array into Call array, so it can be used by the hash functions
-    const callArray: Call[] = callParameters.map((callParameters) => {
-        return {
-            contractAddress: callParameters.toContract.address,
-            entrypoint: callParameters.functionName,
-            calldata: callParameters.toContract.adaptInput(
-                callParameters.functionName,
-                callParameters.calldata
-            )
-        };
-    });
-
-    const executeCallArray: executeCallParameters[] = [];
-    let rawCalldata: RawCalldata = [];
-
-    // Parse the Call array to create the objects which will be accepted by the contract
-    callArray.forEach((call) => {
-        executeCallArray.push({
-            to: BigInt(call.contractAddress),
-            selector: hash.starknetKeccak(call.entrypoint),
-            data_offset: rawCalldata.length,
-            data_len: call.calldata.length
-        });
-        rawCalldata = rawCalldata.concat(call.calldata);
-    });
-
-    const adaptedNonce = nonce.toString();
-    const adaptedMaxFee = "0x" + maxFee.toString(16);
-    const adaptedVersion = "0x" + version.toString(16);
-    const messageHash = hash.hashMulticall(
-        accountAddress,
-        callArray,
-        adaptedNonce,
-        adaptedMaxFee,
-        adaptedVersion
-    );
-
-    const args = {
-        call_array: executeCallArray,
-        calldata: rawCalldata,
-        nonce: adaptedNonce
-    };
-
-    return { messageHash, args };
-}
-
 export async function handleAccountContractArtifacts(
     accountType: string,
     artifactsName: string,
+    artifactsVersion: string,
     hre: HardhatRuntimeEnvironment
 ): Promise<string> {
     // Name of the artifacts' parent folder
     const artifactsBase = artifactsName + ".cairo";
 
-    const baseArtifactsPath = path.join(
-        hre.config.paths.starknetArtifacts,
-        ACCOUNT_CONTRACT_ARTIFACTS_ROOT_PATH
-    );
-
-    // Remove old versions from the path
-    if (fs.existsSync(baseArtifactsPath)) {
-        const contents = fs.readdirSync(baseArtifactsPath);
-        contents
-            .filter((content) => content !== ACCOUNT_ARTIFACTS_VERSION)
-            .forEach((content) => {
-                fs.rmSync(path.join(baseArtifactsPath, content), {
-                    recursive: true,
-                    force: true
-                });
-            });
-    }
+    const baseArtifactsPath = path.join(hre.config.paths.starknetArtifacts, ACCOUNT_ARTIFACTS_DIR);
 
     // Full path to where the artifacts will be saved
     const artifactsTargetPath = path.join(
```

Comment:
Why are artifacs being saved to a target directory?

Reply:
Accounts are also contracts. So for the plugin to be able to interact with them as contracts, their artifacts need to be in the artifacts directory. Internally, doing `account.call(...)` does something like `account.starknetContract.call(...)`